models/new_model

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+#
+# DART software - Copyright UCAR. This open source software is provided
+# by UCAR, "as is", without charge, subject to all terms of use at
+# http://www.image.ucar.edu/DAReS/DART/DART_download
+
+model_name=$1
+location_mod=$2
+
+if [[ ("$1" = "help") || ("$1" = "--help") || ($# -eq 0) ]]; then
+  echo "Usage: $0 [model_name] [location_mod]"
+  exit 0
+elif [[ -d "$model_name" ]]; then
+  echo "$model_name already exists! Please try a different model name."
+  exit 1
+elif [[ $# -eq 1 ]]; then
+  echo "Please enter a proper location! (Hint: check DART/assimilation_code/location for location modules avaliable.)"
+  exit 1
+elif [[ ! -d "../assimilation_code/location/$location_mod" ]]; then
+  echo "$location_mod is not a proper location module! Please try a different location name."
+  exit 1
+else
```

Comment:
Functionally this is good, but I think it is worth refactoring to make the script easier to add to in the future.

Here you have a giant if statement that runs from line 10 to line 48, but you do not need to:

pseudo code:
```
check the arguments
   if arguments are wrong, exit

create the template
```

The argument checking is good, I would create a function "usage" which prints out the usage.